### PR TITLE
feat: combine wm and wg, refactor cc

### PR DESCRIPTION
**Pull request by lekhovitsky**

### CODE REVIEW

File: contracts/support/WithdrawalManagerV3.sol [hunk 1]
```diff
@@ -88,15 +105,25 @@ contract WithdrawalManagerV3 is IWithdrawalManagerV3, ACLTrait, ContractsRegiste
     }
 
     /// @dev Sends all `account`'s immediately withdrawable balance of `token` to `to`
-    function _claimImmediateWithdrawal(address account, address token, address to) internal {
+    function _claimImmediateWithdrawal(address account, address token, address to, bool unwrapWETH) internal {
```
Comment (0xmikko) [inline contracts/support/WithdrawalManagerV3.sol:115]: Original idea was to add address 0xeeeee... for ETH to remove annoying flag convert to ETH

File: contracts/credit/CreditManagerV3.sol [hunk 2]
```diff
@@ -1124,22 +1119,26 @@ contract CreditManagerV3 is ICreditManagerV3, SanityCheckTrait, ReentrancyGuardT
...
+    /// @param convertToETH If true, WETH will be transferred to withdrawal manager, from which Credit Facade can
+    ///        claim it as ETH later in the transaction (ignored if token is not WETH)
     function _safeTokenTransfer(address creditAccount, address token, address to, uint256 amount, bool convertToETH)
         internal
     {
         if (convertToETH && token == weth) {
-            ICreditAccountBase(creditAccount).transfer({token: token, to: wethGateway, amount: amount}); // U:[CM-31, 32]
-            IWETHGatewayV3(wethGateway).deposit({to: to, amount: amount}); // U:[CM-31, 32]
+            ICreditAccountBase(creditAccount).transfer({token: token, to: withdrawalManager, amount: amount}); // U:[CM-31, 32]
+            IWithdrawalManagerV3(withdrawalManager).addImmediateWithdrawal({
```
Comment (0xmikko) [inline contracts/credit/CreditManagerV3.sol]: Please, add a helper function for addImmediteWithdrawal for contract size reduction

File: contracts/credit/CreditConfiguratorV3.sol [hunk 3]
```diff
@@ -156,19 +153,19 @@ contract CreditConfiguratorV3 is ICreditConfiguratorV3, ACLNonReentrantTrait {
...
             emit SetPriceOracle(CreditManagerV3(creditManager).priceOracle()); // I:[CC-1A]
 
-            /// Sets the max debt per block multiplier
-            /// This parameter determines the maximal new debt per block as a factor of
-            /// maximal Credit Account debt - essentially a cap on the number of new Credit Accounts per block
-            _setMaxDebtPerBlockMultiplier(uint8(DEFAULT_LIMIT_PER_BLOCK_MULTIPLIER)); // I:[CC-1]
+            // Sets the max debt per block multiplier
+            // This parameter determines the maximal new debt per block as a factor of
+            // maximal Credit Account debt - essentially a cap on the number of new Credit Accounts per block
+            _setMaxDebtPerBlockMultiplier(address(_creditFacade), uint8(DEFAULT_LIMIT_PER_BLOCK_MULTIPLIER)); // I:[CC-1]
```
Comment (0xmikko) [inline contracts/credit/CreditConfiguratorV3.sol:165]: What the motivation to add creditFacade as parameter?

Comment (lekhovitsky) [reply contracts/credit/CreditConfiguratorV3.sol:165]: In the original implementation, `creditFacade()` was called multiple times in every function, which added around 4-5K gas to the `setCreditFacade`

File: contracts/support/WithdrawalManagerV3.sol (hunk 1 above)
Comment (lekhovitsky) [reply contracts/support/WithdrawalManagerV3.sol:115]: Well, yeah, that's neater, will do

File: contracts/credit/CreditManagerV3.sol (hunk 2 above)
Comment (lekhovitsky) [reply contracts/credit/CreditManagerV3.sol]: Done

File: contracts/support/WithdrawalManagerV3.sol [hunk 4]
```diff
@@ -88,12 +90,17 @@ contract WithdrawalManagerV3 is IWithdrawalManagerV3, ACLTrait, ContractsRegiste
...
-    function claimImmediateWithdrawal(address token, address to, bool unwrapWETH)
+    function claimImmediateWithdrawal(address token, address to)
         external
         override
         nonZeroAddress(to) // U:[WM-4A]
     {
-        _claimImmediateWithdrawal({account: msg.sender, token: token, to: to, unwrapWETH: unwrapWETH});
+        _claimImmediateWithdrawal({
+            account: msg.sender,
+            token: token == ETH_ADDRESS ? weth : token,
```
Comment (0xmikko) [inline contracts/support/WithdrawalManagerV3.sol:100]: Is it correct, it seems that token balance will be stored under ETH_ADDRESS index in map, and it's good to separate it with WETH (user could also have some WETH before). My suggestion is to keep balance under ETH_ADDRESS  as token, and withdraw WETH and send ETH in `_claim...` function

Comment (lekhovitsky) [reply contracts/support/WithdrawalManagerV3.sol:100]: `ETH_ADDRESS` only matters during claim, `addImmediateWithdrawal` and mapping are unaware of it

**Review by 0xmikko — APPROVED** [review]

File: contracts/support/WithdrawalManagerV3.sol (hunk 4 above)
Comment (0xmikko) [reply contracts/support/WithdrawalManagerV3.sol:100]: Got your point, I though it's added in CM under ETH_ADDRESS

**Pull request merged**